HumanDynamicsEstimationLibrary/utils/src/iDynTreeUtils.cpp

```diff
@@ -391,8 +391,15 @@ void state::Integrator::integrate(const iDynTree::VectorDynSize& new_dot_s, cons
 void state::Integrator::integrate(const iDynTree::VectorDynSize& new_dot_s,
                                   const iDynTree::Vector3& new_dot_W_p_B,
                                   const iDynTree::Vector3& new_omega_B,
-                                  const double dt)
+                                  const double dt,
+                                  bool* resetFlag)
 {
+    if(*resetFlag) {
```

Comment:
This will crash if `resetFlag` is a `nullptr`, should we account for this? Is this really required or can we just use a boolean instead of a pointer?

Reply:
Just FYI, not a big fan of boolean arguments (see https://abseil.io/tips/94) but I think this is not blocking the PRs.

Reply:
It is required for it to to be a pointer since it is set `true` in `HumanStateProvider.cpp` and `false` in here.

So, any suggestion is more than welcomed

Reply:
Sorry, I am not sure I got this, if something is boolean it can be both true or false, we can align in person.

Reply:
We sorted this in person: it was necessary to modify the boolean, so it is possible to use the reference instead of the pointer.